### PR TITLE
Refactor: add unit tests and docstrings to remaining components

### PR DESCRIPTION
**Pull request by muselesscreator**
**TL;DR -** 
Adds in unit tests and docstrings for the remaining components.  Very minor updates to individual components for testing.

JIRA: [AU-3](https://openedx.atlassian.net/browse/AU-3)

**Developer Checklist**
- [ ] Test suites passing
- [ ] Documentation and test plan updated, if applicable
- [ ] Received code-owner approving review
- [ ] Bumped version number [package.json](../package.json)

**Testing Instructions**
Smoke test

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [x] I've done a visual code review
- [x] I've tested the new functionality

FYI: @edx/masters-devs-gta

### CODE REVIEW

**Review by jnlapierre — APPROVED** [review]
LGTM

**Review by jnlapierre — APPROVED** [review]
LGTM

**Review by mattcarter — APPROVED** [review]
G2G. Approved.
